independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/options/BootstrapMavenOptions.java

```diff
@@ -101,16 +110,92 @@
     }
 
     public static BootstrapMavenOptions newInstance() {
-        return newInstance(getMavenCmdLine());
+        return newInstance(getMavenCmdLine(false).cmdLine);
     }
 
     public static BootstrapMavenOptions newInstance(String cmdLine) {
         return new BootstrapMavenOptions(parse(cmdLine));
     }
 
-    public static String getMavenCmdLine() {
+    public static class MavenCmdInfo {
+        public final String cmdLine;
+        public final List<String> userProps;
+    
+        public MavenCmdInfo(String cmdLine, List<String> userProps) {
+            this.cmdLine = cmdLine;
+            this.userProps = userProps;
+        }
+    }
+    
+    public static MavenCmdInfo getMavenCmdLine(boolean getUserProps) {
         final String mvnCmd = PropertyUtils.getProperty(QUARKUS_INTERNAL_MAVEN_CMD_LINE_ARGS);
-        return mvnCmd == null ? System.getenv(MAVEN_CMD_LINE_ARGS) : mvnCmd;
+        if (mvnCmd == null || getUserProps) {
+            final List<String> props = updateMavenCmdLineAndExtractUserProperties(System.getenv(MAVEN_CMD_LINE_ARGS), System.getProperties());
+            return new MavenCmdInfo(System.getProperty(QUARKUS_INTERNAL_MAVEN_CMD_LINE_ARGS), props);
+        }
+        return new MavenCmdInfo(mvnCmd, Collections.emptyList());
+    }
+
+    static List<String> updateMavenCmdLineAndExtractUserProperties(String currentCmdLine,
+            Map<Object, Object> properties) {
```

Comment:
TBH, I don't like this method. I don't think we should be updating anything. We need the proper command line string which we can resolve and simply return. IMO, that'll be enough.

Reply:
This method only exists so that we can test the behavior by passing arbitrary command line strings and property maps.

Reply:
Actually, just realized that there already is a `BootstrapMavenOptionsTest` but in a different package. Why is it done like this? Traditionally, tests are put in the same package as classes they test…

Reply:
I'll move all the tests to the same package as the code then, if you don't mind